Makefile.am

```diff
@@ -225,60 +244,78 @@ endif
 PKCS11_USER_PIN ?= 01234567
 PKCS11_VHSM_PIN ?= 1234567890
+if AIX
+PKCSLIB = '@libdir@/opencryptoki/libopencryptoki.a(libopencryptoki.so.0)'
+else
+PKCSLIB ?= @libdir@/opencryptoki/libopencryptoki.so
+endif
+
 
 ci-prepare:
-	killall -HUP pkcsslotd || true
+if AIX
+		chmod u+x $(KILLALL)
+endif
+	$(KILLALL) -HUP pkcsslotd || true
 	${srcdir}/testcases/ciconfig.sh "$(sysconfdir)/opencryptoki" "$(sysconfdir)/opencryptoki" "$(sysconfdir)/opencryptoki"
 	@sbindir@/pkcsslotd
 	@sbindir@/pkcsstats --reset-all
-	for slot in `awk '/slot (.*)/ { print $$2; }' $(sysconfdir)/opencryptoki/opencryptoki.conf`; do @sbindir@/pkcsconf -c $$slot -t | grep "Flags:" | grep -q TOKEN_INITIALIZED || PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so ${srcdir}/testcases/init_token.sh $$slot; done
+	for slot in `awk '/slot (.*)/ { print $$2; }' $(sysconfdir)/opencryptoki/opencryptoki.conf`; do @sbindir@/pkcsconf -c $$slot -t | grep "Flags:" | grep -q TOKEN_INITIALIZED || PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCSLIB=$(PKCSLIB) ${srcdir}/testcases/init_token.sh $$slot; done
 #	cd ${srcdir}/testcases && PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCS11_VHSM_PIN=$(PKCS11_VHSM_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so ./init_vhsm.exp 42
 #	echo "VHSM_MODE" >> "$(sysconfdir)/opencryptoki/ep11tok42.conf"
 
-installcheck-local: all
-	killall -HUP pkcsslotd || true
+installcheck-local: all check util-checks
+if AIX
+		chmod u+x $(KILLALL)
+endif
+	$(KILLALL) -HUP pkcsslotd || true
 	@sbindir@/pkcsslotd
 	if test ! -z ${PKCS11_TEST_USER}; then				\
 		chmod 777 ${srcdir}/testcases &&			\
 		cd ${srcdir}/testcases &&                               \
-		su ${PKCS11_TEST_USER} -s /bin/bash -c "PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so sh ./ock_tests.sh || true"; \
+		su ${PKCS11_TEST_USER} -s /bin/bash -c "PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=$(PKCSLIB) sh ./ock_tests.sh || true"; \
 	else								\
 		cd ${srcdir}/testcases && 				\
-		PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so sh ./ock_tests.sh || true; \
+		PKCS11_SO_PIN=$(PKCS11_SO_PIN) PKCS11_USER_PIN=$(PKCS11_USER_PIN) PKCSLIB=$(PKCSLIB) ./ock_tests.sh || true; \
 	fi
-	killall -HUP pkcsslotd
+	$(KILLALL) -HUP pkcsslotd
 
-ci-installcheck: ci-prepare installcheck
-	killall -HUP pkcsslotd || true
-	@sbindir@/pkcsslotd
-	cd ${srcdir}/testcases						\
-	&& export PKCSLIB=@libdir@/opencryptoki/libopencryptoki.so	\
-	&& export PKCS11_USER_PIN=$(PKCS11_USER_PIN)			\
-	&& export PKCS11_SO_PIN=$(PKCS11_SO_PIN)			\
-	&& ./misc_tests/p11sak_test.sh   | tee log-p11sak.txt		\
-	&& ./misc_tests/pkcsconf_test.sh | tee log-pkcsconf.txt
-	@sbindir@/pkcsstats --all
-	killall -HUP pkcsslotd
-	@echo "done"
+ci-installcheck: ci-prepare installcheck util-checks
```

Comment:
Hm, not sure if the order of the prereqs is deterministic, especially with `make -j <n>`.
What in `util-checks` should be executed only AFTER the other two.
It kills the pkcsslotd and restarts, so this can't run concurrent to other places using the pkcsslotd.

Reply:
You're right, there is no ordering in the target dependency order. Reverted this change.